pkg/db/client.go

```diff
@@ -57,36 +57,49 @@ type (
-	Exec(ctx context.Context, query string, args ...any) (sql.Result, error)
-	NamedExec(ctx context.Context, query string, arg any) (sql.Result, error)
-	ExecMultiInTx(ctx context.Context, sqlers ...Sqler) (results []sql.Result, err error)
-	BindNamed(query string, arg any) (string, []any, error)
-	Prepare(ctx context.Context, query string) (*sql.Stmt, error)
-	Preparex(ctx context.Context, query string) (*sqlx.Stmt, error)
-	PrepareNamed(ctx context.Context, query string) (*sqlx.NamedStmt, error)
-	Query(ctx context.Context, query string, args ...any) (*sql.Rows, error)
-	Queryx(ctx context.Context, query string, args ...any) (*sqlx.Rows, error)
-	QueryRow(ctx context.Context, query string, args ...any) *sql.Row
-	NamedQuery(ctx context.Context, query string, arg any) (*sqlx.Rows, error)
-	Select(ctx context.Context, dest any, query string, args ...any) error
-	NamedSelect(ctx context.Context, dest any, query string, arg any) error
-	Get(ctx context.Context, dest any, query string, args ...any) error
-	WithTx(ctx context.Context, ops *sql.TxOptions, do func(ctx context.Context, tx *sqlx.Tx) error) error
-	Close() error
-}
+type (
+	Client interface {
+		GetSingleScalarValue(ctx context.Context, query string, args ...any) (int, error)
+		GetResult(ctx context.Context, query string, args ...any) (*Result, error)
+		Exec(ctx context.Context, query string, args ...any) (sql.Result, error)
+		NamedExec(ctx context.Context, query string, arg any) (sql.Result, error)
+		ExecMultiInTx(ctx context.Context, sqlers ...Sqler) (results []sql.Result, err error)
+		BindNamed(query string, arg any) (string, []any, error)
+		Prepare(ctx context.Context, query string) (*sql.Stmt, error)
+		Preparex(ctx context.Context, query string) (*sqlx.Stmt, error)
+		PrepareNamed(ctx context.Context, query string) (*sqlx.NamedStmt, error)
+		Query(ctx context.Context, query string, args ...any) (*sql.Rows, error)
+		Queryx(ctx context.Context, query string, args ...any) (*sqlx.Rows, error)
+		QueryRow(ctx context.Context, query string, args ...any) *sql.Row
+		NamedQuery(ctx context.Context, query string, arg any) (*sqlx.Rows, error)
+		Select(ctx context.Context, dest any, query string, args ...any) error
+		NamedSelect(ctx context.Context, dest any, query string, arg any) error
+		Get(ctx context.Context, dest any, query string, args ...any) error
+		WithTx(ctx context.Context, ops *sql.TxOptions, do func(ctx context.Context, tx *sqlx.Tx) error) error
+		Close() error
+	}
 
-type ClientSqlx struct {
-	logger   log.Logger
-	db       *sqlx.DB
-	executor exec.Executor
-}
+	ClientSqlx struct {
+		logger   log.Logger
+		db       *sqlx.DB
+		executor exec.Executor
+	}
 
-type clientCtxKey string
+	clientSqlxOptions struct {
+		executorOptions []exec.BackoffExecutorOption
+		clientOption    []func(*ClientSqlx)
+	}
 
-func ProvideClient(ctx context.Context, config cfg.Config, logger log.Logger, name string) (Client, error) {
+	ClientSqlxOption func(*clientSqlxOptions)
```

Comment:
This is a client option, right? The sqlx part sounds like an implementation detail